compiler/ast.nim

```diff
@@ -225,6 +225,7 @@ type
     nkBreakState,         # special break statement for easier code generation
     nkFuncDef,            # a func
     nkTupleConstr         # a tuple constructor
+    nkEarlySemArg         # When an argument was semmed early, but we might need the untyped AST later, the semmed AST is the first, the unsemmed the second son
```

Comment:
Simply remembering both variants can create a compiler that uses an expontential amount of memory. (Nest these calls to get 2*2*2*2*... memory consumption.) There should only be a single AST that works with all the constructs. My "semtityped" AST idea.

Reply:
No worries, I have a fix for the exponential growth

Reply:
*Now* I worry...

Reply:
We just discard our reference to the unsemmed/semmed AST when choosing which one we need after sigmatch. https://github.com/nim-lang/Nim/pull/14863/commits/fb0e697bbdf20a4087dfd463ab018d2fbb1f683e